.pre-commit-config.yaml

```diff
@@ -26,10 +26,15 @@ repos:
                 files: python/.*\.(py|pyx|pxd)$
                 types: [file]
       - repo: https://github.com/pre-commit/mirrors-mypy
-        rev: 'v0.782'
+        rev: 'v0.971'
         hooks:
               - id: mypy
-                args: ["--config-file=setup.cfg", "python/cudf/cudf", "python/dask_cudf/dask_cudf", "python/custreamz/custreamz", "python/cudf_kafka/cudf_kafka"]
+                additional_dependencies: [types-cachetools]
```

Comment:
In light of #11661 is this necessary, or can we wait to introduce this until we have a more holistic strategy for which dependencies get included here?

Reply:
This is a necessary dependency. Without it:
```
$ pre-commit run --all-files mypy
python/cudf/cudf/utils/cudautils.py:5: error: Library stubs not installed for "cachetools" (or incompatible with Python 3.9)
python/cudf/cudf/core/udf/utils.py:5: error: Library stubs not installed for "cachetools" (or incompatible with Python 3.9)
python/cudf/cudf/core/udf/utils.py:5: note: Hint: "python3 -m pip install types-cachetools"
python/cudf/cudf/core/udf/utils.py:5: note: (or run "mypy --install-types" to install all missing stub packages)
python/cudf/cudf/core/udf/utils.py:5: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```